bookkeeper-server/src/test/java/org/apache/bookkeeper/client/GenericEnsemblePlacementPolicyTest.java

```diff
@@ -132,9 +132,9 @@ public void testReplaceBookie() throws Exception {
                     lh.addEntry(value);
                 }
             }
-            assertEquals(2, customMetadataOnNewEnsembleStack.size());
```

Comment:
@eolivelli I don't know what others will be impacted. I do not see such binary compatibility issue a lot, because typically when a new version is available, most of the people would update their application with a newer version and compile it. this would be detected as compilation issues and fixed. 

Compilation issues are much safer and easier to deal with. In contrast, I did see a lot of runtime exceptions when newer clients uses old dependencies (when there aren't any troubles compiling new clients with old dependencies), e.g. method signatures are changed, method not found, classes not found. Those runtime exceptions are much dangerous than compilation issues. 

>> are you ok with closing the issue as won't fix

I am fine with that.